added sign in with apple (#86)
go-pkgz/auth: README.md

## Register oauth2 providers

Authentication handled by external providers. You should setup oauth2 for all (or some) of them to allow users to authenticate. It is not mandatory to have all of them, but at least one should be correctly configured.

> #### Apple Auth Provider

Reviewer: ```suggestion

#### Apple Auth Provider
```

For configure this provider user require Apple developer account (without it setting up sign in with Apple impossible). [Sign in with Apple](https://developer.apple.com/documentation/sign_in_with_apple/sign_in_with_apple_rest_api) lets users log in to your app using their two-factor authentication Apple ID.

Follow to next steps for configure on Apple side: 

> 1. At first you need login in to developer account and enable capabilities `Sign in with Apple" for specific App ID (in Certificates, Identifiers & Profiles section ).  Also you need `App ID Prefix` as TeamID value, it's required for configure Apple auth provider.

Reviewer: Is it possible to add links for instruction steps?

Author: links added

`PrivateKeyLoaderInterface` used to [implement](https://github.com/zebox/auth/blob/master/provider/apple.go#L98:L100) custom loader for private key (which was generated on step 4) used for create `client_secret`. 
Also, user can use pre-defined function `provider.LoadApplePrivateKeyFromFile(filePath string)` for load private key from local file.

`AddAppleProvide` try to load private key at call and return error if load failed. Always check error when call this provider.    
  
```go
    if err := service.AddAppleProvider("apple", appleCfg, provider.LoadApplePrivateKeyFromFile(os.Getenv("AEXMPL_APPLE_PRIVKEY_PATH"))); err != nil {
		log.Fatalf("[ERROR] create AppleProvider failed: %v", err)
	}
```
**Limitation:** 
* Map a userName (if specific scope defined) can be only at first login for a user. 
Every next login with Apple, under account which sign in early, will no return field with userName and provider can't fetch user name until a user delete sign in for you service with Apple ID in  Apple account profile (security section).
Provider always get user `UID` (`sub` claim) and `email` (if email scope defined) from claims of `IDToken`.

> * Apple hasn't API for fetch avatar and user info. 

Reviewer: ```suggestion
* Apple doesn't have an API for fetch avatar and user info. 
```

> See [example](https://github.com/zebox/auth/blob/master/_example/main.go#L83:L93) before use.

Reviewer: ```suggestion
See [example](https://github.com/go-pkgz/auth/blob/master/_example/main.go#L83:L93) before use.
```

5. You should make sure, that SPF (DNS record) for a domain (which used for `Sign in with Apple` and bind to `Service ID`) is verified. It's can be done it in `Certificates, Identifiers & Profiles` >> [`More`](https://developer.apple.com/account/resources/services/configure) section.

After complete previous steps you can go to configure `Apple auth provider`. For it require parameters which must set in `AppleConfig`:

* _ClientID_ (**required**) - Service ID identifier which use for `Sign with Apple`
* _TeamID_ (**required**) - Identifier a developer account (use as prefix for all App ID)
* _KeyID_ (**required**) - Identifier a generated key for `Sign with Apple`
* _Scopes_ (optional) - The [scope](https://developer.apple.com/documentation/sign_in_with_apple/clientconfigi/3230955-scope) is the amount of user information requested from Apple. Right now, there are only two options name and email. You can request the user’s name or email. You can also choose to request both or neither.
* _ClientSecretExpire_ (optional) - time in seconds when client secret will expire (default: 1800 secs = 30 minutes)

```go 
    // apple config parameters  

> 	appleCfg := provider.AppleConfig{

Reviewer: If some keys are required, your code must explicitly fail when they are not provided.

Reviewer: Nvm, I see that it's doing the check already.
